File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/entity/Invoice.java
package org.mifos.mobilewallet.core.data.fineract.entity;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by ankur on 07/June/2018
 */

public class Invoice implements Parcelable {

    public static final Creator<Invoice> CREATOR = new Creator<Invoice>() {
        @Override
        public Invoice createFromParcel(Parcel in) {
            return new Invoice(in);
        }

        @Override
        public Invoice[] newArray(int size) {
            return new Invoice[size];
        }
    };
    @SerializedName("consumerId")
    String consumerId;
    @SerializedName("consumerName")
    String consumerName;
    @SerializedName("amount")
    double amount;
    @SerializedName("itemsBought")
    String itemsBought;
    @SerializedName("status")
    long status;
    @SerializedName("transactionId")
    String transactionId;
    @SerializedName("id")
    long id;
    @SerializedName("title")
    String title;
    @SerializedName("date")
    List<Integer> date = new ArrayList<>();

    public Invoice(String consumerId, String consumerName, double amount, String itemsBought,
            long status, String transactionId, long id, String title, List<Integer> date) {
        this.consumerId = consumerId;
        this.consumerName = consumerName;
        this.amount = amount;
        this.itemsBought = itemsBought;
        this.status = status;
        this.transactionId = transactionId;
        this.id = id;
        this.title = title;
        this.date = date;
    }

    protected Invoice(Parcel in) {
        this.consumerId = in.readString();
        this.consumerName = in.readString();
        this.amount = in.readLong();
        this.itemsBought = in.readString();
        this.status = in.readInt();
        this.transactionId = in.readString();
        this.id = in.readInt();
        this.title = in.readString();
        this.date = new ArrayList<Integer>();
        in.readList(this.date, Integer.class.getClassLoader());
    }

    public String getConsumerId() {
        return consumerId;
    }

    public void setConsumerId(String consumerId) {
        this.consumerId = consumerId;
    }

    public String getConsumerName() {
        return consumerName;
    }

    public void setConsumerName(String consumerName) {
        this.consumerName = consumerName;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public String getItemsBought() {
        return itemsBought;
    }

    public void setItemsBought(String itemsBought) {
        this.itemsBought = itemsBought;
    }

    public long getStatus() {
        return status;
    }

    public void setStatus(long status) {
        this.status = status;
    }

    public String getTransactionId() {
        return transactionId;
    }

    public void setTransactionId(String transactionId) {
        this.transactionId = transactionId;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }


    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.consumerId);
        dest.writeString(this.consumerName);
        dest.writeDouble(this.amount);
        dest.writeString(this.itemsBought);
        dest.writeLong(this.status);
        dest.writeString(this.transactionId);
        dest.writeList(this.date);
        dest.writeString(this.title);
    }

    @Override
    public String toString() {
        return "Invoice{" +
                "consumerId='" + consumerId + '\'' +
                ", consumerName='" + consumerName + '\'' +
                ", amount=" + amount +
                ", itemsBought='" + itemsBought + '\'' +
                ", status=" + status +
                ", transactionId='" + transactionId + '\'' +
                ", title=" + title +
                ", id=" + id +
                ", date=" + date +
                '}';
    }

    public List<Integer> getDate() {
        return date;
    }

    public void setDate(List<Integer> date) {
        this.date = date;
    }
}

File: core/model/src/main/java/com/mifospay/core/model/entity/Invoice.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-wallet/blob/master/LICENSE.md
 */
package com.mifospay.core.model.entity

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import kotlinx.parcelize.RawValue

@Parcelize
data class Invoice(

    @SerializedName("consumerId")
    var consumerId: String? = null,

    @SerializedName("consumerName")
    var consumerName: String? = null,

    @SerializedName("amount")
    var amount: Double = 0.0,

    @SerializedName("itemsBought")
    var itemsBought: String? = null,

    @SerializedName("status")
    var status: Long = 0L,

    @SerializedName("transactionId")
    var transactionId: String? = null,

    @SerializedName("id")
    var id: Long = 0L,

    @SerializedName("title")
    var title: String? = null,

    @SerializedName("date")
    var date: @RawValue MutableList<Int> = ArrayList(),

) : Parcelable {
    constructor() : this(null, null, 0.0, null, 0L, null, 0L, null, ArrayList())
}
